tests/core/fsck-on-boot/task.yaml

```diff
@@ -76,12 +97,14 @@ execute: |
       # chain.
       unmount_vfat
       cat /proc/self/mountinfo >boot1-after-umount.log
-      dd if=/dev/sda2 of=dirty skip=65 bs=1 count=1 conv=notrunc
+      if os.query is-core16 || os.query is-core18; then
+        dd if=/dev/disk/by-label/system-boot of=dirty skip=65 bs=1 count=1 conv=notrunc
+      elif os.query is-core20; then
+        dd if=/dev/disk/by-label/ubuntu-seed of=dirty skip=65 bs=1 count=1 conv=notrunc
+      fi
       test "$(od -t x1 -A n dirty)" = " 00"  # NOTE: the leading space is relevant
 
       # Reboot to restore mount points.
       REBOOT
       ;;
   esac
-restore: |
-  tests.cleanup restore
```

Comment:
why is this removed? this seems necessary for the test to run given that we use `tests.cleanup defer ...` and if the test fails before executing the `tests.cleanup pop` then things will not be cleaned up properly

Reply:
this was a mistake, the idea was to move it to the top of the file.